source/index.ts

```diff
@@ -1,5 +1,12 @@
 import * as util from 'util';
 
+type TypedArray = Int8Array | Uint8Array | Uint8ClampedArray | Int16Array | Uint16Array | Int32Array | Uint32Array | Float32Array | Float64Array;
```

Comment:
if we do it like 

```
type TypedArray = Array<Int8Array | Uint8Array | Uint8ClampedArray | Int16Array | Uint16Array | Int32Array | Uint32Array | Float32Array | Float64Array>;
```

then typescript will not raise 'cannot find signature on union types' for calling normal functions like `reduce`, `map`, `indexOf` etc.

Reply:
if you want, i can add a commit to this.

Reply:
Although it fixes the issue, the type is incorrect. Now `TypedArray` is defined as an array of `Int8Array` or `Uint8Array`, etc.

What does work however, is the following

```ts
type TypedArray = (Int8Array | Uint8Array | Uint8ClampedArray | Int16Array | Uint16Array | Int32Array | Uint32Array | Float32Array | Float64Array) & Array<number>;
```

This is ugly though. Will have a look as to why this error is caused.

Reply:
AFAIK, the error message "Cannot invoke an expression whose type lacks a call signature"" is by design. see microsoft/typescript#7294 .

Reply:
I think I'm going to leave it as is for now. It's explained in the issue that you referenced, this is by design. If you know what type it will be, it would be safer to use `is.int8Array` or one of the others. Thanks for letting me know, learned something new today :).